### PR TITLE
Added iOS Gateway Example

### PR DESCRIPTION
**Pull request by galz10**

### CODE REVIEW

**Comment by galz10** [comment]
@gguuss

File: examples/complex/esp32/ios-gateway/ios-gateway.ino [hunk 1]
```diff
@@ -0,0 +1,197 @@
+#if defined(ARDUINO_SAMD_MKR1000) or defined(ESP8266)
```
Comment (gguuss) [inline examples/complex/esp32/ios-gateway/ios-gateway.ino:1]: Apache header

File: src/jwt.cpp [hunk 2]
```diff
@@ -24,7 +24,7 @@
 static const String base64_chars =
     "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
     "abcdefghijklmnopqrstuvwxyz"
-    "0123456789+/";
```
Comment (gguuss) [inline src/jwt.cpp:27]: I think we want to keep the fixes that Alvaro patched in, I recommend removing this from your PR.

File: examples/complex/esp32/ios-gateway/extra/Podfile [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# Uncomment the next line to define a global platform for your project
```
Comment (gguuss) [inline examples/complex/esp32/ios-gateway/extra/Podfile:1]: Arduino specification says it's `extras` folder not `extra` folder.

File: examples/complex/esp32/ios-gateway/extra/iOS_IoTCore_Client.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 4]
```diff
@@ -0,0 +1,79 @@
+{
```
Comment (gguuss) [inline examples/complex/esp32/ios-gateway/extra/iOS_IoTCore_Client.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:1]: I do not think you should be checking in files that are generated by the build step, remove this file.

File: examples/complex/esp32/ios-gateway/extra/iOS_IoTCore_Client.xcodeproj/project.xcworkspace/xcuserdata/danilo.xcuserdatad/IDEFindNavigatorScopes.plist [hunk 5]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<array/>
+</plist>
```
Comment (gguuss) [inline examples/complex/esp32/ios-gateway/extra/iOS_IoTCore_Client.xcodeproj/project.xcworkspace/xcuserdata/danilo.xcuserdatad/IDEFindNavigatorScopes.plist:5]: I cannot comment on the binary personal data file you checked in (galzahavi.xcuserdata) but this and other files specific to your account should not be checked in.

File: examples/complex/esp32/ios-gateway/ios-gateway.ino (hunk 1 above)
Comment (gguuss) [inline examples/complex/esp32/ios-gateway/ios-gateway.ino:1]: Apache header missing

File: src/jwt.cpp [hunk 6]
```diff
@@ -24,7 +24,7 @@
 static const String base64_chars =
```
Comment (gguuss) [inline src/jwt.cpp:24]: I don't think we want to pave over the fixes from Alvaro in master, remove this file

File: examples/complex/esp32/ios-gateway/extra/iOS_IoTCore_Client_Demo/SwiftUIToUIKitViewController.swift [hunk 7]
```diff
@@ -0,0 +1,27 @@
+//
+//  SwiftUIToUIKitViewController.swift
+//  UIKitSwiftUIDemo
+//
+//  Created by Danilo Campos on 6/16/20.
+//  Copyright © 2020 Danilo Campos. See license file.
```
Comment (gguuss) [inline examples/complex/esp32/ios-gateway/extra/iOS_IoTCore_Client_Demo/SwiftUIToUIKitViewController.swift:6]: Just checking, what's the license?

File: examples/complex/esp32/ios-gateway/extra/iOS_IoTCore_Client.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved (hunk 4 above)
Comment (gguuss) [inline examples/complex/esp32/ios-gateway/extra/iOS_IoTCore_Client.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:1]: Remove this file (it's generated by cocoapods IIRC)

File: src/jwt.cpp (hunk 2 above)
Comment (galz10) [reply src/jwt.cpp:27]: yea i'll change it to Alvaros i thought i changed it on my fork but i guess not.

File: examples/complex/esp32/ios-gateway/extra/iOS_IoTCore_Client_Demo/SwiftUIToUIKitViewController.swift (hunk 7 above)
Comment (galz10) [reply examples/complex/esp32/ios-gateway/extra/iOS_IoTCore_Client_Demo/SwiftUIToUIKitViewController.swift:6]: It didn't have a license but i didn't use any code from this file i deleted it before pushing it

**Comment by galz10** [comment]
I removed the jwt.cpp and jwt.h

**Pull request closed without merging**